src/utils/SpokeUtils.ts

```diff
@@ -62,15 +53,14 @@ export async function getBlockRangeForDepositId(
   // queriedIds cache to see if the deposit ID at the block number has already been queried. If not, it will
   // make an eth_call request to get the deposit ID at the block number. It will then cache the deposit ID
   // in the queriedIds cache.
-  const _getDepositIdAtBlock = async (blockNumber: number): Promise<number> => {
-    queriedIds[blockNumber] ??= await spokePool._getDepositIdAtBlock(blockNumber);
-    return queriedIds[blockNumber];
+  const getDepositIdAtBlock = async (blockTag: number): Promise<number> => {
+    return queriedIds[blockTag] ??= await spokePool.numberOfDeposits({ blockTag });
```

Comment:
Intuitively this implementation is wrong to me, since numberOfDeposits != "highest depositId". I'd expect that we should be doing something like:

```
const nDeposits = await spokePoolClient.numberOfDeposits({ blockTag });
return Math.max(nDeposits - 1, 0);
```

nb. the above ignores the in-memory cache, but it's implied that that is used.